config.go

```diff
@@ -474,8 +478,15 @@ func loadConfig() (*config, []string, error) {
 		pLog.Warnf("%v", configFileError)
 	}
 
+	// Ensure a domain is set if HTTPS via letsencrypt is preferred.
+	if cfg.UseLEHTTPS && cfg.Domain == "" {
+		return nil, nil, fmt.Errorf("a valid domain is required for HTTPS " +
+			"via letsencrypt")
+	}
+
 	// Generate the TLS cert and key if they not already exist.
-	if !fileExists(defaultTLSCertFile) || !fileExists(defaultTLSKeyFile) {
+	if cfg.UseLEHTTPS && (!fileExists(defaultTLSCertFile) ||
```

Comment:
Is this correct? if UseLEHTTPS then generate a self-signed cert pair? Should this be `!UseLEHTTPS`?

Reply:
you're right, good catch.